117-v1-CRD-API-Conversion-Tool.md

```diff
@@ -0,0 +1,137 @@
+# `v1` CRD API Conversion Tool
+
+This proposal covers `v1` CRD API Conversion Tool which will be available to the users to help them with the migration from the `v1beta2` APIs to the `v1` API.
+
+## Motivation
+
+The migration of the Strimzi CRs from the `v1beta2` version to the `v1` version can be done manually.
+We plan to cover the manual process as part of our documentation.
+However, given the amount of steps and in some cases large number of resources that would need to be converted, we should also provide a Conversion Tool to help with it.
+That was also approved as part of the initial [Strimzi `v1` CRD API and 1.0.0 release](./113-Strimzi-v1-CRD-API-and-1.0.0-release.md) proposal.
+This proposal provides additional details.
+
+## Proposal
+
+### Conversion Tasks
+
+The conversion tool should be able to fulfill the following tasks:
+* Support conversion of the custom resources by helping modifying their structure to adjust the fields that were renamed or moved:
+    * Be able to do the conversion inside a Kubernetes cluster in all namespace, in a single namespace, or a particular custom resource only
+    * Convert YAML files with the Strimzi custom resources in-place (by modifying the existing file) or by creating a new file
+* Support updating the CRDs:
+    * Change the storage version to `v1` in the CRD `.spec` section
+    * "Touch" all custom resources to make sure they are stored under the new version
+    * Change the stored versions in the CRD `.status` section in order to allow the old versions to be removed
+
+Users should be able to use only some of these tasks if needed and combine the Conversion Tool and the manual conversion process as needed.
+
+### Architecture
+
+The Conversion Tool will be written in Java and mostly follow the architecture of the previous tool we used last time when migrating from `v1beta1` to `v1beta2` API.
+Using Java would simplify the release process as it would depend directly on the Strimzi API.
+
+The Conversion Tool itself will be distributed as ZIP and TAR.GZ archives.
+The archives will contain helper scripts for running it from the command line.
+It will use Java 17 as most of the other Strimzi components.
+
+It will be also bundled in the Strimzi Operator container.
```

Comment:
Would it make sense to have simple separate image just for the conversion tool? I mean, it would be more lightweight possibly?

Reply:
A separate image would be smaller. So I guess it would be more _lightweight_. But the overall release process will be more heavyweight as new image means certainly more work and time compared to another thing added to the existing image. 

Also, the conversion tool seems to have roughly ~20MB. Given the operator image has IIRC 400-500MB and most of it is the base OS, I do not think it makes sense to have a separate image.